numba_cuda/numba/cuda/dispatcher.py

```diff
@@ -158,6 +159,16 @@ def link_to_library_functions(library_functions, library_path,
 
         self.maybe_link_nrt(link, tgt_ctx, asm)
 
+        for k, v in cres.fndesc.typemap.items():
+            if not isinstance(v, Function):
+                continue
+
+            if not isinstance(v.typing_key, ExternFunction):
+                continue
+
+            for obj in v.typing_key.link:
+                lib.add_linking_file(obj)
```

Comment:
I'm trying to understand the code flow of this part a bit deeper. Is NVRTC going to be invoked on every `CUSource` addition?

Reply:
I believe so - is there another way?

Reply:
Nah - It seems like this is the only way. I think at best what we can do is to cache each CUSource and do some source code analysis to determine which source can be fused and compile in the same pass. Though that's too heavy handed and may actually bring more overhead.